src/native-common/View.tsx

```diff
@@ -343,17 +343,20 @@ export class View extends ViewBase<Types.ViewProps, Types.Stateless> {
             accessibilityComponentType: AccessibilityUtil.accessibilityComponentTypeToString(props.accessibilityTraits),
             accessibilityLiveRegion: AccessibilityUtil.accessibilityLiveRegionToString(props.accessibilityLiveRegion)
         };
-        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation && (this.props.onPress ||
-                (this.props.tabIndex !== undefined && this.props.tabIndex >= 0))) {
+
+        if (_isNativeMacOs && App.supportsExperimentalKeyboardNavigation) {
             const macAccessibilityProps: MacComponentAccessibilityProps = accessibilityProps as any;
-            if (this.props.tabIndex !== -1) {
+
+            if (this.props.tabIndex && this.props.tabIndex >= 0) {
```

Comment:
This isn't correct. If tabIndex is 0, this condition will fail and the code below is not run
If tab index is undefined and onPress is defined, the code below should be run but is not

The conditional that was previously above is needed on this block, so I don't think it really optimizes anything by moving it

Reply:
ok, tabindex: -2